velox/exec/tests/RowContainerTest.cpp

```diff
@@ -32,44 +32,149 @@ using namespace facebook::velox::test;
 
 class RowContainerTest : public exec::test::RowContainerTestBase {
  protected:
-  void testExtractColumnForOddRows(
+  void testExtractColumn(
       RowContainer& container,
       const std::vector<char*>& rows,
       int column,
-      const VectorPtr& expected) {
+      const VectorPtr& expected,
+      int offset = 0) {
+    // This set of tests uses all variations of the extractColumn API to copy
+    // column value for all rows in the container.
+    testExtractColumnAllRows(container, rows, column, expected, offset);
+    // This set of tests use extractColumn with an out-of-order pattern
+    // and repetitions.
+    testExtractColumnOutOfOrder(container, rows, column, expected, offset);
+  }
+
+  void testExtractColumnAllRows(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected,
+      int offset = 0) {
     auto size = rows.size();
 
-    // Extract only odd rows.
-    std::vector<char*> oddRows(size, nullptr);
-    for (auto i = 1; i < size; i += 2) {
-      oddRows[i] = rows[i];
-    }
+    auto testBasic = [&](vector_size_t offset = 0) -> void {
```

Comment:
Let's remove default value for offset parameter and pass it explicitly. This would make it easier to read the code.

Drop -> void.